integration_tests/src/main/python/asserts.py

```diff
@@ -277,10 +277,12 @@ def _assert_gpu_and_cpu_writes_are_equal(
     if current_test_has_delta_marker() and not current_test_allows_non_gpu_delta_write():
         print("Delta Lake test detected - applying Delta write validation")
         from delta_lake_utils import assert_rapids_delta_write
-        assert_rapids_delta_write(lambda spark: write_func(spark, gpu_path), conf=conf)
+        gpu_result = assert_rapids_delta_write(lambda spark: write_func(spark, gpu_path), conf=conf)
     else:
-        with_gpu_session(lambda spark : write_func(spark, gpu_path), conf=conf)
+        gpu_result = with_gpu_session(lambda spark : write_func(spark, gpu_path), conf=conf)
     gpu_end = time.time()
+    if gpu_result is not None and cpu_result is not None:
```

Comment:
Do we need to check the corner case that only one of them is None?

Reply:
Ah actually you are right. I will fix this condition.

Reply:
OK, I just removed this condition. `assert_equal` should handle all edge cases. Thanks for catching this.